R/reverse_operons.R

```diff
@@ -57,14 +69,28 @@ straightenOperonSeq <- function(prot) {
 
 ## The function to reverse operons
 
-#' reverseOperonSeq
+#' reverseOperon: Reverse the Direction of Operons in Genomic ContextSeq
+#'
+#' @description
+#' This function processes a genomic context data frame to reverse the direction
+#' of operons based on specific patterns in the GenContext column. It handles 
+#' elements represented by ">" and "<" and restructures the genomic context by 
+#' flipping the direction of operons while preserving the relationships 
+#' indicated by "=".
+#'
+#' @param prot [data.frame] A data frame containing at least a column named 
+#' 'GenContext', which represents the genomic contexts that need to be reversed.
 #'
-#' @param prot
+#' @return [data.frame] The input data frame with the 'GenContext' column updated t
+#' o reflect the reversed operons.
 #'
-#' @return
 #' @export
 #'
 #' @examples
+#' # Example genomic context data frame
+#' prot <- data.frame(GenContext = c("A>B", "C<D", "E=F*G", "H>I"))
+#' reversed_prot <- reverseOperonSeq(prot)
+#' print(reversed_prot)
```

Comment:
@awasyn , can you double check this example code?

Reply:
Thanks. I will  run example again.

Reply:
Done.

Reply:
I'm still seeing an R-CMD check error when running this example:
```
❯ checking examples ... [14s/15s] ERROR
  Running examples in ‘MolEvolvR-Ex.R’ failed
  The error most likely occurred in:
  
  > base::assign(".ptime", proc.time(), pos = "CheckExEnv")
  > ### Name: reverseOperonSeq
  > ### Title: reverseOperon: Reverse the Direction of Operons in Genomic
  > ###   ContextSeq
  > ### Aliases: reverseOperonSeq
  > 
  > ### ** Examples
  > 
  > # Example genomic context data frame
  > prot <- data.frame(GenContext = c("A>B", "C<D", "E=F*G", "H>I"))
  > reversed_prot <- reverseOperonSeq(prot)
  Error in ge[[x]] : subscript out of bounds
  Calls: reverseOperonSeq -> lapply -> FUN -> straightenOperonSeq
  Execution halted
```

Using the example data defined in `prot` the error occurs during the lapply operation [@line 137]( https://github.com/awasyn/MolEvolvR/blob/cb76c69eba5586c255834a370bc7ffa035700b8c/R/reverse_operons.R#L137) as `ge` has length 0 due to the previous subset operation.